numpy/core/src/multiarray/datetime.c

```diff
@@ -1761,6 +1762,9 @@ parse_datetime_unit_from_string(char *str, Py_ssize_t len, char *metastr)
                 return NPY_FR_as;
         }
     }
+    else if (allow_generic && len == 7 && !strncmp(str, "generic", 7)) {
```

Comment:
It seemed there was probably a reason to disallow `m8[generic]` instead of `m8`. In any case, it didn't sound like a good idea to make parsing more lenient in a bugfix release.